example/counterBasedRng/src/counterBasedRng.cpp

```diff
@@ -0,0 +1,258 @@
+            Key<TAcc> const& key,
+            Vec<TAcc> const& threadElemExtent,
+            Vec<TAcc>& threadFirstElemIdx) -> void
+        {
+            auto const threadLastElemIdx = threadFirstElemIdx[I] + threadElemExtent[I];
+            auto const threadLastElemIdxClipped
+                = (dst.extents[I] > threadLastElemIdx) ? threadLastElemIdx : dst.extents[I];
+
+            constexpr auto Dim = alpaka::Dim<TAcc>::value;
+
+            const auto firstElem = threadFirstElemIdx[I];
+            if constexpr(I < Dim - 1)
+            {
+                for(; threadFirstElemIdx[I] < threadLastElemIdxClipped; ++threadFirstElemIdx[I])
+                {
+                    ElemLoop<I + 1>::elemLoop(acc, dst, key, threadElemExtent, threadFirstElemIdx);
+                }
+            }
+            else
+            {
+                Counter<TAcc> c = {0, 0, 0, 0};
+                for(unsigned int i = 0; i < Dim; ++i)
+                    c[i] = threadFirstElemIdx[i];
+
+                for(; threadFirstElemIdx[Dim - 1] < threadLastElemIdxClipped; ++threadFirstElemIdx[Dim - 1])
+                {
+                    c[Dim - 1] = threadFirstElemIdx[Dim - 1];
+                    const auto random = Gen<TAcc>::generate(c, key);
+                    // to make use of the whole random vector we would need to ensure numElement[0] % 4 == 0
+                    dst[threadFirstElemIdx] = TElem(random[0]);
+                }
+            }
+            threadFirstElemIdx[I] = firstElem;
+        }
+    };
+
+public:
+    //! The kernel entry point.
+    //!
+    //! \tparam TAcc The accelerator environment to be executed on.
+    //! \tparam TElem The matrix element type.
+    //! \param acc The accelerator to be executed on.
+    //! \param dst destimation matrix.
+    //! \param extent The matrix dimension in elements.
+    ALPAKA_NO_HOST_ACC_WARNING
+    template<typename TAcc, typename TElem>
+    ALPAKA_FN_ACC auto operator()(
+        TAcc const& acc,
+        alpaka::experimental::BufferAccessor<TAcc, TElem, alpaka::Dim<TAcc>::value, alpaka::experimental::WriteAccess>
+            dst,
+        Key<TAcc> const& key) const -> void
+    {
+        constexpr auto Dim = alpaka::Dim<TAcc>::value;
+        static_assert(Dim <= 4, "The CounterBasedRngKernel expects at most 4-dimensional indices!");
+
+        Vec<TAcc> const gridThreadIdx(alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc));
+        Vec<TAcc> const threadElemExtent(alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc));
+        Vec<TAcc> threadFirstElemIdx(gridThreadIdx * threadElemExtent);
+
+        ElemLoop<0>::elemLoop(acc, dst, key, threadElemExtent, threadFirstElemIdx);
+    }
+};
+
+auto main() -> int
+{
+// Fallback for the CI with disabled sequential backend
+#if defined(ALPAKA_CI) && !defined(ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED)
+    return EXIT_SUCCESS;
+#else
```

Comment:
We cannot build/run this example in the CI with other backends? why?

Reply:
I think this is the problem we have with all examples. The example is using the hard-coded serial accelerator, therefore this acc is required.

Reply:
It is expected to be run with another backend enabled, but the serial one is to serve as the host reference, and thus must also be enabled. Technically, any CPU backend could play this role, but I do not know of a utility we have to select any CPU enabled backend (and fail if none are enabled). `alpaka::ExampleDefaultAcc<Dim, Idx>;` is used as the other, the device, backend.